tools/build_pak_files.py

```diff
@@ -0,0 +1,129 @@
+
+        for content_dir in content_dirs:
+
+            # remove existing Content dir
+            unreal_project_content_dir = os.path.join(unreal_project_dir, "Content")
+            if spear.path_exists(unreal_project_content_dir):
+                print(f"[SPEAR | build_pak_files.py] Symlink exists, removing: {unreal_project_content_dir}")
+                spear.remove_path(unreal_project_content_dir)
+
+            # create symlink
+            print(f"[SPEAR | build_pak_files.py] Creating symlink: {unreal_project_content_dir} -> {content_dir}")
+            os.symlink(content_dir, unreal_project_content_dir)
+
+            for platform in args.platforms:
+
+                cmd = [
+                    unreal_editor_bin,
+                    uproject,
+                    "-run=Cook",
+                    "-TargetPlatform=" + platform + "NoEditor",
+                    "-fileopenlog",
+                    "-ddc=InstalledDerivedDataBackendGraph",
+                    "-unversioned",
+                    "-stdout",
+                    "-FullStdOutLogOutput",
+                    "-CrashForUAT",
+                    "-unattended",
+                    "-NoLogTimes",
+                    "-UTF8Output"
+                ]
+                print(f"[SPEAR | build_pak_files.py] Executing: {' '.join(cmd)}")
+                cmd_result = subprocess.run(cmd, check=True)
+
+                platform_dir = os.path.realpath(os.path.join(unreal_project_dir, "Saved", "Cooked", f"{platform}NoEditor"))
+                content_dirs = [
+                    os.path.join(platform_dir, "Engine", "Content", "Animation"),
+                    os.path.join(platform_dir, "Engine", "Content", "BasicShapes"),
+                    os.path.join(platform_dir, "Engine", "Content", "EngineResources"),
+                    os.path.join(platform_dir, "Engine", "Content", "Functions", "Engine_MaterialFunctions02", "Math"),
+                    os.path.join(platform_dir, "Engine", "Content", "Functions", "Engine_MaterialFunctions02", "Texturing"),
+                    os.path.join(platform_dir, "Engine", "Content", "Functions", "Engine_MaterialFunctions02", "Utility"),
+                    os.path.join(platform_dir, "Engine", "Content", "Functions", "MaterialLayerFunctions"),
+                    os.path.join(platform_dir, project,  "Content")
+                ]
+
+                # create the output_dir
+                os.makedirs(args.output_dir, exist_ok=True)
+
+                file_name = f"Map_{os.path.basename(content_dir)}_{platform}"
+
+                # construct path to the output pak file
+                pak_file = os.path.join(args.output_dir, file_name + ".pak")
+
+                # text file used to generate the final pak file
+                txt_path = os.path.join(args.output_dir, file_name + ".txt")
+
+                for i, content_dir in enumerate(content_dirs):
```

Comment:
This inner `content_dir` variable masks an existing `content_dir` variable. Is this intended?

Reply:
Updated code to avoid this conflict.